src/app/messages-page/messages-page.component.ts

```diff
@@ -33,4 +36,37 @@ export class MessagesPageComponent {
     }
     this.showThreadView = true;
   }
+
+  _toggleSettingsTray() {
+    this.globalVars.openSettingsTray = !this.globalVars.openSettingsTray;
+  }
+
+  _settingsTrayBeOpen() {
+    return this.globalVars.openSettingsTray;
+  }
+
+  // This marks all messages as read and relays this request to the server.
+  _markAllMessagesReadMobile() {
+    for (let thread of this.globalVars.messageResponse.OrderedContactsWithMessages) {
+      this.globalVars.messageResponse.UnreadStateByContact[thread.PublicKeyBase58Check] = false;
+    }
```

Comment:
if the endpoint fails, we'll have shown the user that all the threads were marked as read.  Is that okay?